include/circt/Dialect/FIRRTL/InnerSymbolTable.h

```diff
@@ -22,6 +22,56 @@
 namespace circt {
 namespace firrtl {
 
+/// The target of an inner symbol, the entity the symbol is a handle for.
+class InnerSymTarget {
+public:
+  /// Default constructor, invalid.
+  /* implicit */ InnerSymTarget() { assert(!*this); }
+
+  /// Target an operation, and optionally a field (=0 means the op itself).
+  explicit InnerSymTarget(Operation *op, size_t fieldID = 0)
+      : op(op), portIdx(invalidPort), fieldID(fieldID) {}
+
+  /// Target a port, and optionally a field (=0 means the port itself).
+  /// Operation should be an FModuleLike.
+  explicit InnerSymTarget(size_t portIdx, Operation *op, size_t fieldID = 0)
+      : op(op), portIdx(portIdx), fieldID(fieldID) {}
+
+  /// Create a target for a field, given a target to a base.
+  explicit InnerSymTarget(const InnerSymTarget &base, size_t fieldID)
+      : op(base.op), portIdx(base.portIdx), fieldID(fieldID) {
+    assert(base.fieldID == 0);
```

Comment:
Is there a good reason to require `base.fieldID == 0`? Also, nice use of `const &`.

Reply:
Not especially, I just like assert'ing out cases I hadn't thought out thoroughly.  So, thanks for bringing it up :+1: .

The thinking was this would "only" be used to create field targets given a base (fieldID=0) target, hence the check on valid use.  But that doesn't really seem an important or necessary restriction -- question is if the fieldID should be set absolutely or relative to the specified target (since that's just addition)?  They're the same when `base.fieldID=0`, the latter seems more generally useful but maybe is a bit unexpected.  Perhaps this would make more sense as static method instead of copy-ish constructor, and can clarify the behavior via method name to avoid the ambiguity.